share/functions/help.fish

```diff
@@ -23,58 +22,58 @@ function help --description 'Show help for the fish shell'
 	# by the help function defined below.
 	#
 	set -l graphical_browsers htmlview x-www-browser firefox galeon mozilla konqueror epiphany opera netscape rekonq google-chrome chromium-browser
-	set -l text_browsers htmlview www-browser links elinks lynx w3m
 
-	if type -q "$BROWSER"
-		# User has manually set a preferred browser, so we respect that
-		set fish_browser $BROWSER
-
-		# If browser is known to be graphical, put into background
-		if contains -- $BROWSER $graphical_browsers
-			set fish_browser_bg 1
-		end
+	if type -q "$fish_help_browser[1]"
```

Comment:
Is fish_help_browser guaranteed to have at least one element at this point? Because if not (and I don't see any such guarantee) this will fail. Also, it's not obvious to me that the code you based this on should be using `type -q`. If the user has set fish_help_browser we should try to use it regardless of whether or not `type -q` thinks it is a valid command. I recommend just doing `if set -q fish_help_browser[1]` (not the lack of a dollar-sign).